themes/default/content/blog/pulumi-release-notes-m64/index.md

```diff
@@ -0,0 +1,238 @@
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Pulumi;
+using ElasticCloud = Pulumi.ElasticCloud;
+
+class Program
+{
+    static Task Main() =>
+        Deployment.Run(() => {
+            var latest = Output.Create(ElasticCloud.GetStack.InvokeAsync(new ElasticCloud.GetStackArgs
+            {
+                Region = "us-east-1",
+                VersionRegex = "latest",
+            });
+
+            var deployment = ElasticCloud.Deployment("my-deployment", new ElasticCloud.DeploymentArgs
+            {
+                Region = "us-east-1",
+                Version = latest.Version,
+                DeploymentTemplateId = "aws-io-optimized-v2",
+                ElasticSearch = {},
+            });
+        });
+}
+```
+
+{{% /choosable %}}
+
+{{< /chooser >}}
+
+### AWS Native provider supports auto-naming
+
+Most Pulumi providers support [auto-naming for resources]({{<relref "/docs/intro/concepts/resources#autonaming">}}), which makes it easier to have multiple stacks of the same Pulumi program by avoiding naming conflicts, among other benefits. Now, our AWS Native provider joins the club with full auto-naming support.
+
+[Learn more about AWS auto-naming in this GitHub issue](https://github.com/pulumi/pulumi-aws-native/issues/156)
+
+### New resources in the AWS Native provider
+
+We shipped new versions of the AWS Native provider (0.3.0 through 0.4.0) that added [38 new resources](https://github.com/pulumi/pulumi-aws-native/compare/v0.2.0...v0.4.0#diff-1ac835cc58d7899e9299c7570151c7b0d7732c78f1bd53fe25fd4189b72e168e) giving you access to resources like `AWS::EKS::Cluster`, `AWS::Lightsail::Database`, and several new resources in the `AWS::Redshift` namespace.
+
+### New resources in the Azure Native provider
+
+We shipped new versions of the Azure Native provider (1.43.0 through 1.45.0) that collectively added [4 new resources](https://github.com/pulumi/pulumi-azure-native/blob/master/CHANGELOG.md#1450-2021-11-05) giving you access to resources like Cognitive Services.
+
+## Pulumi CLI and core technologies
+
+In this milestone, we shipped Pulumi versions [3.16.1](https://github.com/pulumi/pulumi/releases/tag/v3.16.1) through [3.17.0](https://github.com/pulumi/pulumi/releases/tag/v3.17.0). The full list of changes in each version is available in the [changelog](https://github.com/pulumi/pulumi/blob/master/CHANGELOG.md); read on to learn about some of the biggest changes.
+
+### Functions now support outputs
+
+When creating complex infrastructure, it's common to pass information about one resource to a function provided by another resource. Previously, doing this in Pulumi required you to use `apply` to get the information from the first resource and pass it into the function. Now, we've simplified the process so you can reference outputs directly from the function—no `apply` needed. To take advantage of these changes, make sure you've updated to the latest versions of Pulumi and the providers you're using.
```

Comment:
`function provided by another resource` --> perhaps just "to a function". It does not seem to me functions are associated with or provided by resources, perhaps only very indirectly (as in, some resources are always used with some functions together in practice).